src/Faithlife.WebRequests/Json/ExplicitStatusResponse.cs

```diff
@@ -0,0 +1,32 @@
+using System.Net;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace Faithlife.WebRequests.Json
+{
+	/// <summary>
+	/// AutoWebServiceRequest responses with an explicitly set status code. 
+	/// </summary>
+	public abstract class ExplicitStatusResponse : AutoWebServiceResponse
+	{
+		/// <summary>
+		/// Exposes the response status code.
+		/// </summary>
+		public HttpStatusCode StatusCode { get; internal set; } // internal instead of private to avoid being compiled away
```

Comment:
What does this comment mean?

I have not seen the compiler remove private setters from the generated code just because they're not used; [example](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQGMD2A7ZMFQCW2MyAPgAIDMABBQEw0DCNA3gLABQNPdtFARgAMNAAoJMABwFsA5hBgBuGpISEAbrAg1kCxQF8uvPnWFiJkxqxrylNYjAgJsUADY69NQ531A=).

Reply:
This was puzzling me too; I can set this `private` in LinqPad and the code works, but in Toby's environment the `AutoWebServiceRequest` was unable to write to a `private` version of this setter with reflection.

I admit I didn't verify this particular explanation for why it didn't work for him.

Reply:
Ah! Yes, the `private set` is actually problematic when trying to get properties for the derived type (such as `EchoResponseExplicitStatusResponse` used for the tests).

```
PropertyInfo statusCodeProperty = GetProperty(responseType.GetBaseType(), nameof(HttpResponseMessage.StatusCode));
if (statusCodeProperty != null)
...
```
would negate the need for an `internal set`. I'm not sure it's worth that, though; I prefer the consistency in `AutoWebServiceRequest`.

Reply:
Inheritance shouldn't affect `internal` vs `private`, unless I'm misunderstanding something fundamental.

Try this in LinqPad:

```
 void Main()
{
	var property = GetProperty(typeof(Foo), "Bar").Dump();
	var frob = new Frob();
	property.SetValue(frob, "test", null);
	frob.Dump();
}

// Define other methods and classes here
public class Foo
{
	public string Bar { get; private set; }
}

public class Frob : Foo
{
	
}

private static PropertyInfo GetProperty(Type type, string propertyName)
{
	return type.GetRuntimeProperties().FirstOrDefault(x => x.GetMethod != null && !x.GetMethod.IsStatic && string.Equals(x.Name, propertyName, StringComparison.OrdinalIgnoreCase));
}
```

Reply:
I learned a thing today! This makes more sense now, nice sleuthing Toby.

`internal` seems fine, with an updated comment.

Reply:
Comment updated. 9c6a8a47ed60beacfa362fcb6831fa078b832668

Reply:
Better yet, we could [implement the fix here](https://stackoverflow.com/questions/31807874/cant-get-getsetmethod-for-property-with-private-setter-from-base-class).

```csharp
private static PropertyInfo GetProperty(Type type, string propertyName)
{
	var property = type.GetRuntimeProperties().FirstOrDefault(x => x.GetMethod != null && !x.GetMethod.IsStatic && string.Equals(x.Name, propertyName, StringComparison.OrdinalIgnoreCase));
	if (property.SetMethod == null)
		property = property.DeclaringType.GetRuntimeProperty(property.Name);
	return property;
}
```

Reply:
We could do that, we'd just also need to remove ` && statusCodeProperty.CanWrite` from [line 84.](https://github.com/Faithlife/FaithlifeWebRequests/pull/11/files#diff-ab8b513c7bdc4b22952e6c53a828f3c7R84)

Reply:
Why would we need to modify line 84?

Reply:
>Why would we need to modify line 84?

You're right; I was mistaken in thinking `private set` also mean `CanWrite` would be false.